storage/Devices/BcacheImpl.cc

```diff
@@ -415,7 +536,20 @@ namespace storage
 
 	SystemCmd cmd(cmd_line, SystemCmd::DoThrow);
 
-	sleep(1);
+        // TODO: is it enough? check with Coly how to recognize it
+	sleep(1); // give bcache some time to finish its async operation, so bcache device exists
+
+	cmd_line = ECHO_BIN " " + quote(toString(cache_mode)) + " > " + quote(SYSFS_DIR "/" + get_sysfs_path() + "/bcache/cache_mode");
+
+	SystemCmd cmd2(cmd_line, SystemCmd::DoThrow);
+
+	cmd_line = ECHO_BIN " " + to_string(writeback_percent) + " > " + quote(SYSFS_DIR "/" + get_sysfs_path() + "/bcache/writeback_percent");
+
+	SystemCmd cmd3(cmd_line, SystemCmd::DoThrow);
+
+	cmd_line = ECHO_BIN " " + to_string(sequential_cutoff) + " > " + quote(SYSFS_DIR "/" + get_sysfs_path() + "/bcache/sequential_cutoff");
+
+	SystemCmd cmd4(cmd_line, SystemCmd::DoThrow);
```

Comment:
Setting the attributes here is fine. But it will not allow to just change the attributes without creating the bcache. For that new actions (or one big action) are required, see e.g. Gpt::Impl::add_modify_actions().

For the time being a TODO notice is fine for me.

Reply:
OK, for now it is a bit intention as with modify there is a problem with that async operations, which is not needed for cache_mode, but others will need at first detach, then modify and then attach again ( source Coly, but I found on network, then e.g. writeback_percent should be possible to modify without detach, so still something that needs to be make clear ).

So I will add that TODO.

Reply:
@jreidinger sorry, the grammar is so bad that I don't understand what you mean. can you say that again, more carefully, please?

examples: modify != modification, modifying; detach != detaching; is != are; then != than != that

Reply:
@mvidner OK. So basically for now we support only add/remove bcache device. Changing configuration for existing bcache device is problematic as some operations require detach of cache, which is expensive operation.